docs/spec/lightclient/Lightclient_A_1.tla

```diff
@@ -0,0 +1,463 @@
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+LightStoreUpdateStates(statuses, ht, blockState) ==
+    [h \in DOMAIN statuses \union {ht} |->
+        IF h = ht THEN blockState ELSE statuses[h]]      
+
+\* Check, whether newHeight is a possible next height for the light client.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+ScheduleTo(newHeight, pNextHeight, pTargetHeight, pLatestVerified) ==
+    LET ht == pLatestVerified.header.height IN
+    \/ /\ ht = pNextHeight
+       /\ ht < pTargetHeight
+       /\ pNextHeight < newHeight
+       /\ newHeight <= pTargetHeight
+    \/ /\ ht < pNextHeight
+       /\ ht < pTargetHeight
+       /\ ht < newHeight
+       /\ newHeight < pNextHeight
+    \/ /\ ht = pTargetHeight
+       /\ newHeight = pTargetHeight
+
+\* The loop of VerifyToTarget.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+VerifyToTargetLoop ==
+      \* the loop condition is true
+    /\ latestVerified.header.height < TARGET_HEIGHT
+      \* pick a light block, which will be constrained later
+    /\ \E current \in BC!LightBlocks:
+        \* Get next LightBlock for verification
+        /\ IF nextHeight \in DOMAIN fetchedLightBlocks
+           THEN /\ LightStoreGetInto(current, nextHeight)
+                /\ UNCHANGED fetchedLightBlocks
+           ELSE /\ FetchLightBlockInto(current, nextHeight)
+                /\ fetchedLightBlocks' = LightStoreUpdateBlocks(fetchedLightBlocks, current)
+        \* Record that one more probe has been done (for complexity and model checking)
+        /\ nprobes' = nprobes + 1
+        \* Verify the current block
+        /\ LET verdict == ValidAndVerified(latestVerified, current) IN
+           \* Decide whether/how to continue
+           CASE verdict = "OK" ->
+              /\ lightBlockStatus' = LightStoreUpdateStates(lightBlockStatus, nextHeight, "StateVerified")
+              /\ latestVerified' = current
+              /\ state' =
+                    IF latestVerified'.header.height < TARGET_HEIGHT
+                    THEN "working"
+                    ELSE "finishedSuccess"
+              /\ \E newHeight \in HEIGHTS:
+                 /\ ScheduleTo(newHeight, nextHeight, TARGET_HEIGHT, current)
+                 /\ nextHeight' = newHeight
+                  
+           [] verdict = "CANNOT_VERIFY" ->
+              (*
+                do nothing: the light block current passed validation, but the validator
+                set is too different to verify it. We keep the state of
+                current at StateUnverified. For a later iteration, Schedule
+                might decide to try verification of that light block again.
+                *)
+              /\ lightBlockStatus' = LightStoreUpdateStates(lightBlockStatus, nextHeight, "StateUnverified")
+              /\ \E newHeight \in HEIGHTS:
+                 /\ ScheduleTo(newHeight, nextHeight, TARGET_HEIGHT, latestVerified)
```

Comment:
..and here: pick a height in (latestVerified, nextHeight-1]
```suggestion
                 /\ ScheduleTo(newHeight, latestVerified, nextHeight-1)
```

Reply:
Josef and I still think that we should not change the number of parameters and the parameters to `CanScheduleTo`, as we may change the scheduling decision in the future.